matrix_client/api.py

```diff
@@ -68,13 +68,37 @@ def __init__(self, base_url, token=None):
         self.validate_cert = True
 
     def initial_sync(self, limit=1):
-        """Perform /initialSync.
+        """ Deprecated. Use sync instead.
+        Perform /initialSync.
 
         Args:
             limit(int): The limit= param to provide.
         """
         return self._send("GET", "/initialSync", query_params={"limit": limit})
 
+    def sync(self, since="", timeout_ms=30000, **kwargs):
+        """ Perform a sync request.
+
+        Args:
+            since(string): Optional. A token which specifies where to continue
+                a sync from.
+            timeout_ms(int): Optional. The time in milliseconds to wait.
+            **kwargs: Specify filter, full_state or set_presence here.
+        """
+        request = {
+            "timeout": timeout_ms
+        }
+
+        if since != "":
+            request["since"] = since
+
+        for i in ["filter", "fullstate", "set_presence"]:
```

Comment:
If kwargs is only allowed some stuff, itd be better to be explicit and check for None's